modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc

```diff
@@ -0,0 +1,113 @@
+[id='proc-sending-requests-to-the-rbac-rest-api-by-using-curl_{context}']
+= Sending requests to the RBAC REST API by using the curl utility
+
+You can send RBAC REST API requests by using the curl utility.
+
+.Prerequisites
+* xref:enabling-and-giving-access-to-rbac[You have access to the RBAC feature].
+
+.Procedure
+include::snip-finding-bearer-token.adoc[]
+
+. In a terminal, run the curl command and review the response:
++
+--
+.Default request
+[source,subs="+attributes,+quotes"]
+----
+curl -v -H "Content-Type: application/json" \
+  -H "Authorization: Bearer _<token>_" \
+  -X __<method>__ "{my-product-url}/__<endpoint>__" \
+----
+
+.POST request requiring JSON body data
+[source,subs="+attributes,+quotes"]
+----
+curl -v -H "Content-Type: application/json" \
+  -H "Authorization: Bearer _<token>_" \
+  -X POST "{my-product-url}/__<endpoint>__" \
+  -d __<body>__
+----
+
+__<token>__::
+Enter your saved authorization token.
+
+__<method>__::
+Enter the HTTP method for your xref:ref-rbac-rest-api-endpoints_{context}[API endpoint].
+
+* `GET`: To retrieve specified information from a specified resource endpoint.
+* `POST`: To create or update a resource.
+* `PUT`: To update a resource.
+* `DELETE`: To delete a resource.
+
+pass:c,a,q[{my-product-url}]::
+Enter your {product-short} URL.
+
+__<endpoint>__::
+Enter the xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] to which you want to send a request, such as `/api/permission/policies`.
+
+__<body>__::
+Enter the JSON body with data that your xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] might need with the HTTP `POST` request.
+
+.Example request to create or update a role
+[source,subs="+attributes,+quotes"]
+----
+curl -v -H "Content-Type: application/json" \
+  -H "Authorization: Bearer _<token>_" \
+  -X POST "{my-product-url}/api/permission/roles" \
+  -d '{
+      "memberReferences": ["group:default/example"],
+      "name": "role:default/test",
+      "metadata": { "description": "This is a test role" }
+    }'
+----
+
+.Example request to create or update a permission policy
+[source,subs="+attributes,+quotes"]
+----
+curl -v -H "Content-Type: application/json" \
+  -H "Authorization: Bearer $token" \
+  -X POST "{my-product-url}/api/permission/policies" \
+  -d '[{
+      "entityReference":"role:default/test",
+      "permission": "catalog-entity",
+      "policy": "read", "effect":"allow"
+    }]'
+----
+
+.Example request to create or update a condition
+[source,subs="+attributes,+quotes"]
+----
+
+curl -v -H "Content-Type: application/json" \
```

Comment:
I agree with 'create'. Regarding 'update', I believe we have a different URL pattern. The user needs to specify a condition ID. The POST method returns the condition ID, and it is also present in the conditions JSON when the user retrieves the list of conditions. Here is an example of the PUT request:

```
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer $token" \
  -X PUT "{my-product-url}/api/permission/roles/conditions/1" \
  -d '{
          "result":"CONDITIONAL",
          "roleEntityRef":"role:default/test",
          "pluginId":"catalog",
          "resourceType":"catalog-entity",
          "permissionMapping": ["read",  "update", "delete"],
          "conditions": {
            "rule": "IS_ENTITY_OWNER",
            "resourceType": "catalog-entity",
            "params": {"claims": ["group:default/janus-authors"]}
          }
       }'
```